src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -103,9 +118,10 @@ public void testDoGet_ValidAdminUser() throws IOException, ServletException {
 
     adminServlet.doGet(mockRequest, mockResponse);
 
+    Mockito.verify(mockRequest).setAttribute("map", map);
     Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
```

Comment:
verify the contents of the map in the response?

Reply:
I was struggling with that, any advice?

Reply:
I think you're on the right track.. you need to mock all the "stores" calls, eg
    Mockito.when(mockUserStore.countTotalUsers()).thenReturn(9);
.. and make your "map" correspond.

Reply:
I tried making all these in the newest commit I just made (I tested literally everything), but I can't make it work, I'm getting a Null Pointer Exception in the method `addStats()` from ´AdminServlet.java´, I tried separating them into different tests, one for `addStats` and another for the `doGet_ValidAdminUser`, but still was getting the Null pointer exception. I'm kind of stuck with this test.
Can I get the approval so that I can start adding a few more stats, and in the future I can make the test work.

Reply:
You're getting the NPE at line 70 I think only because you didn't call 

adminServlet.setMessageStore(mockMessageStore)

in AdminServletTest.setup(), no?

You also need to call adminServlet.setConversationStore().. that's what I meant in my earlier comment

Reply:
Oh my god, I got confused, I thought you said to remove them, figured out that because I wasn't doing a post request I didn't need them :octocat: mybad!

Reply:
I added that and it worked! I'm so happy it did thanks @tucosh ! 😄